staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java

```diff
@@ -282,6 +283,20 @@ public void cleanUpStaleInprogressBatches()
         }
     }
 
+    @Override
+    public BatchStatus getBatchStatus(final TenantId tenantId, final BatchId batchId) throws BatchNotFoundException
+    {
+        if (batchPathProvider.getPathForBatch(tenantId, batchId).toFile().exists()) {
+            return BatchStatus.COMPLETED;
+        } else if (skipBatchFileCleanup && isInProgressBatchAbandoned(batchPathProvider.getTenantInprogressDirectory(tenantId), batchId)) {
```

Comment:
Because only if this flag is set to true, the cleanup batch will not be running. By default, the flag is false, the batch runs and there will not be any abandoned batches.

Reply:
You're assuming there isn't a second Staging Service that has it configured on.  It only takes one Staging Service instance to clean up for all.